### PR TITLE
DRY in tests using linter variable

### PR DESCRIPTION
**Pull request by IndrajeetPatil**

### CODE REVIEW

File: tests/testthat/test-class_equals_linter.R [hunk 1]
```diff
@@ -1,42 +1,48 @@
...
-  expect_lint("identical(class(x), c('glue', 'character'))", NULL, class_equals_linter())
+  expect_lint("identical(class(x), c('glue', 'character'))", NULL, linter)
 })
 
 test_that("class_equals_linter blocks simple disallowed usages", {
+  linter <- class_equals_linter()
+
   expect_lint(
     "if (class(x) == 'character') stop('no')",
     rex::rex("Instead of comparing class(x) with =="),
```
Comment (AshesITR) [inline tests/testthat/test-class_equals_linter.R:16]: Factor out `msg` as well?

File: tests/testthat/test-expect_identical_linter.R [hunk 2]
```diff
@@ -1,55 +1,61 @@
...
+
+  expect_lint("expect_equal(x, 1.034)", NULL, linter)
+  expect_lint("expect_equal(x, c(1.01, 1.02))", NULL, linter)
   # whole numbers with explicit decimals are OK, even in mixed scenarios
-  expect_lint("expect_equal(x, c(1.0, 2))", NULL, expect_identical_linter())
+  expect_lint("expect_equal(x, c(1.0, 2))", NULL, linter)
   # plain numbers are still caught, however
   expect_lint(
     "expect_equal(x, 1L)",
     rex::rex("Use expect_identical(x, y) by default; resort to expect_equal() only when needed"),
```
Comment (AshesITR) [inline tests/testthat/test-expect_identical_linter.R:45]: ditto

File: tests/testthat/test-expect_identical_linter.R [hunk 3]
```diff
@@ -1,55 +1,61 @@
...
-  expect_lint("expect_equal(x, y, check.attributes = FALSE)", NULL, expect_identical_linter())
+  expect_lint("expect_equal(x, y, check.attributes = FALSE)", NULL, linter)
 })
 
 test_that("expect_identical_linter blocks simple disallowed usages", {
+  linter <- expect_identical_linter()
+
   expect_lint(
     "expect_equal(x, y)",
     rex::rex("Use expect_identical(x, y) by default; resort to expect_equal() only when needed"),
```
Comment (AshesITR) [inline tests/testthat/test-expect_identical_linter.R:23]: ditto

**Review by AshesITR — COMMENTED** [review]
Probably can avoid some of the line breaks after refactoring while still staying below maximum linde width.

File: tests/testthat/test-expect_identical_linter.R (hunk 3 above)
Comment (IndrajeetPatil) [reply tests/testthat/test-expect_identical_linter.R]: Done!

File: tests/testthat/test-class_equals_linter.R (hunk 1 above)
Comment (IndrajeetPatil) [reply tests/testthat/test-class_equals_linter.R]: Done!

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**